dcpy/lifecycle/builds/dbt.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+import typer
+from ruamel.yaml import YAML
+
+from dcpy.utils.logging import logger
+from dcpy.models.lifecycle.builds import InputDataset, Recipe
+from dcpy.lifecycle.builds import plan
+
+DEFAULT_RECIPE = "recipe.yml"
+
+app = typer.Typer(add_completion=False)
+
+yaml = YAML()
+
+
+def add_to_recipe_sources(dataset, source):
+    source["tables"].append({"name": dataset})
+    return source
+
+
+class RecipeEditor:
+    def __init__(self, path):
+        self.path = path
+
+    def __enter__(self):
+        plan.recipe_from_yaml(self.path)
+        with self.path.open() as f:
+            self.recipe = yaml.load(f)
+        return self.recipe
+
+    def __exit__(self, *args):
+        Recipe(**self.recipe)
```

Comment:
Since this object doesn't know what you're doing with the recipe, good to just validate it before dumping